urllib3/util/retry.py

```diff
@@ -290,5 +289,17 @@ def __repr__(self):
                     cls=type(self), self=self)
 
 
+def redirect_method(method, status):
```

Comment:
I'm of a few minds on this.

Overriding could be a nice feature, and if we want to do that, then the Retry object (I think as an instance method) would be the place to do it. But, I'm not sure how much actual usage that'd get - does urllib3 have previous issues requesting that feature?

My preferred place would be to put it directly on the `Response` object, but I'm not sure if we can do that without patching into httplib private attributes on the response or passing the request method in anyway.